src/NUnitFramework/tests/Internal/PlatformDetectionTests.cs

```diff
@@ -281,6 +281,15 @@ public void DetectMacOSX()
                 new OSPlatform(PlatformID.MacOSX, new Version(0, 0)),
                 "MacOSX");
         }
+
+#if !PORTABLE
+        [Test]
+        [Platform(Include = "MacOSX")]
+        public void CurrentPlatformIsMacOSX()
+        {
+            Assert.That(OSPlatform.CurrentPlatform.IsMacOSX, Is.True);
```

Comment:
This is a rather lame test for the fix, but it allowed me to put a break point in the code. I think the fixes for #869 are actually better indicators of the fix for #515. All tests are now passing on MacOSX, this one is getting hit and I am waiting on CI to test Linux and Windows.